### PR TITLE
bt-2: create MUI color theme

### LINKED ISSUE #2: Create Material UI theme with our custom colors
Image

Info: https://mui.com/material-ui/customization/color/
Design: https://www.figma.com/file/Yd7USgP7PLvA0W794jHXWu/BSA_recruting?type=design&node-id=0-1&mode=design&t=mvb2tCBVBeWTg5yG-0

### PR DESCRIPTION
**Pull request by Simpsonkk**
Hi! I added custom props in MUI theme but MUI requires interfaces instead of type, can I add /* eslint-disable @typescript-eslint/consistent-type-definitions */ ?

### CODE REVIEW

**Comment by nika-sirak** [comment]
I'm working on the typography part of mui theme. As far as I know, we should have one theme with two properties - palette and typography. Should we separate them into different files?

**Comment by anton-otroshchenko** [comment]
> I'm working on the typography part of mui theme. As far as I know, we should have one theme with two properties - palette and typography. Should we separate them into different files?

In my opinion it is better to create a base theme and extend this theme with other ones. But both variants work well

**Review by anton-otroshchenko — APPROVED** [review]

**Review by JosuerBague — APPROVED** [review]

**Comment by Simpsonkk** [comment]
Last commit is final version

File: frontend/src/bundles/common/components/color-theme/color-theme.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/color-theme/color-theme.ts:1]: color theme is not component

File: frontend/src/bundles/common/components/color-theme/theme.d.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import {
+    type Palette as MuiPalette,
+    type PaletteOptions as MuiPaletteOptions,
+    type TypeBackground as MuiTypeBackground,
+} from '@mui/material/styles';
+
+declare module '@mui/material/styles' {
+    interface TypeBackground extends MuiTypeBackground {
+        primary?: string;
+        secondary?: string;
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/color-theme/theme.d.ts]:
```suggestion
        primary: string;
        secondary: string;
```

File: frontend/src/bundles/common/components/color-theme/theme.d.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        };
+    }
+
+    interface PaletteOptions extends MuiPaletteOptions {
+        input?: {
+            main?: string;
+        };
+        tag?: {
+            main?: string;
+        };
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/components/color-theme/theme.d.ts]:
```suggestion
        input: {
            main: string;
        };
        tag: {
            main: string;
        };
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
Looks like you duplicate you code

File: frontend/src/bundles/common/components/color-theme/color-theme.ts
Comment (Simpsonkk) [reply frontend/src/bundles/common/components/color-theme/color-theme.ts:1]: please look at the last commit. My files are located in frontend/src/bundles/common/themes/
image

File: frontend/src/bundles/common/components/color-theme/theme.d.ts (hunk 1 above)
Comment (Simpsonkk) [reply frontend/src/bundles/common/components/color-theme/theme.d.ts]: corrected

File: frontend/src/bundles/common/components/color-theme/theme.d.ts (hunk 2 above)
Comment (Simpsonkk) [reply frontend/src/bundles/common/components/color-theme/theme.d.ts]: corrected

File: frontend/src/bundles/common/components/color-theme/color-theme.ts
Comment (Simpsonkk) [reply frontend/src/bundles/common/components/color-theme/color-theme.ts:1]: I removed duplicated files from old commits

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]
we should add this colors as css variables to

`assets/css/variables.scss`

also conflicts

**Comment by Simpsonkk** [comment]
> we should add this colors as css variables to
> 
> `assets/css/variables.scss`
> 
> also conflicts

@nikita-remeslov 

I created CSS vars imported in styles.scss used in color theme like this and have error because of  'var(--error-main)' but var(--error-main) without quotes it doesn`t work too.
image
image
image
image

**Comment by Simpsonkk** [comment]
I added color vars and an enum for color-theme.ts
I was trying to use CSS vars in the color theme but the color theme can`t use it
I was trying to implement this but because of linter error I quit it and used enum
https://medium.com/@christian.tonye_16869/scss-variables-in-react-typescript-components-de19d7f96245

I can`t resolve conflicts because if I do that, then a merge will occur, and we are not allowed to do this yet

File: frontend/src/bundles/common/enums/colors.enum.ts
Comment (nikita-remeslov) [inline frontend/src/bundles/common/enums/colors.enum.ts:1]: you should add import here - frontend/src/bundles/common/enums/enums.ts

File: frontend/src/bundles/common/themes/color-theme/color-theme.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import { createTheme } from '@mui/material/styles';
+
+import { Colors } from '../../enums/colors.enum.js';
```
Comment (nikita-remeslov) [inline frontend/src/bundles/common/themes/color-theme/color-theme.ts]:
```suggestion
import { Colors } from '../../enums/enums.js';
```

**Review by nikita-remeslov — CHANGES_REQUESTED** [review]

File: frontend/src/bundles/common/enums/colors.enum.ts
Comment (Simpsonkk) [reply frontend/src/bundles/common/enums/colors.enum.ts:1]: done 
image

File: frontend/src/bundles/common/themes/color-theme/color-theme.ts (hunk 3 above)
Comment (Simpsonkk) [reply frontend/src/bundles/common/themes/color-theme/color-theme.ts]: done
image

**Comment by Simpsonkk** [comment]
I see part of the same colors vars. They in development branch after this task task/bt-10-add-radio-component.

File: frontend/src/assets/css/variables.scss [hunk 4]
```diff
@@ -1,4 +1,25 @@
 :root {
+    // Error colors
+    --error-main: #ff2121;
+
+    // Action colors
+    --action-hover: #2e61ae;
+
+    // Background colors
+    --background-primary: #f7f8fc;
+    --background-secondary: #274f8d;
```
Comment (Monyster) [inline frontend/src/assets/css/variables.scss:10]: This color ( --background-secondary: #274f8d; ) in figma is used mainly to highlight buttons and labels.
You can call it ( --accent-primary: #274f8d; )

image
image

Comment (Simpsonkk) [reply frontend/src/assets/css/variables.scss:10]: I think it's just a background for different components and we don't need to give it a unique name. Will see :)

**Review by nikita-remeslov — APPROVED** [review]

**Pull request merged**